src/test/scala/chiselTests/DirectionalBulkConnectSpec.scala

```diff
@@ -731,44 +731,201 @@ class DirectionalBulkConnectSpec extends ChiselFunSpec with Utils {
+      }
+      class MyModule extends Module {
+        val in  = IO(Flipped(new Decoupled()))
+        val out = IO(new ReadyValid())
+        out :<>= in
+      }
+      val out = ChiselStage.emitChirrtl({ new MyModule() }, true, true)
+      testCheck(out, Seq(
+        """out.valid <= in.valid""",
+        """in.ready <= out.ready""",
+      ), Nil)
+    }
+    it("(?.c) Waiving ok-to-unassign field connecting a narrower bus to a wider bus, with defaults for unassigned fields set via last connect semantics") {
+      class ReadyValid extends Bundle {
+        val valid = Bool()
+        val ready = Flipped(Bool())
+      }
+      class Decoupled extends ReadyValid {
+        val data = Waivable(UInt(32.W), false, true)
+      }
+      class MyModule extends Module {
+        val in  = IO(Flipped(new ReadyValid()))
+        val out = IO(new Decoupled())
+        out :<= (new Decoupled()).Lit(_.data.underlying -> 0.U, _.valid -> false.B, _.ready -> false.B)
+        (out: ReadyValid) :<>= in
       }
       val out = ChiselStage.emitChirrtl({ new MyModule() }, true, true)
       testCheck(out, Seq(
-        """wire w0 : { necessary : { foo : UInt<3>, flip bar : UInt<3>}, optional : { foo : UInt<3>, flip bar : UInt<3>}}""",
-        """w0.optional.bar <= UInt<2>("h3")""",
-        """w0.optional.foo <= UInt<2>("h2")""",
-        """w0.necessary.bar <= UInt<1>("h1")""",
-        """w0.necessary.foo <= UInt<1>("h0")""",
-        """wire w1 : { necessary : { foo : UInt<3>, flip bar : UInt<3>}}""",
-        """w1.necessary.bar <= UInt<1>("h1")""",
-        """w1.necessary.foo <= UInt<1>("h0")""",
-        """w0.necessary.bar <= w1.necessary.bar""",
-        """w1.necessary.foo <= w0.necessary.foo""",
+        """out.data <= UInt<1>("h0")""",
+        """out.valid <= UInt<1>("h0")""",
+        """in.ready <= out.ready""",
+        """out.valid <= in.valid""",
       ), Nil)
     }
+    it("(?.d) Waiving ok-to-unassign field connecting a narrower bus to a wider bus will error if no default specified") {
+      class ReadyValid extends Bundle {
+        val valid = Bool()
+        val ready = Flipped(Bool())
+      }
+      class Decoupled extends ReadyValid {
+        val data = Waivable(UInt(32.W), true, true)
+      }
+      class MyModule extends Module {
+        val in  = IO(Flipped(new ReadyValid()))
+        val out = IO(new Decoupled())
+        (out: ReadyValid) :<>= in
+      }
+      intercept[Exception] {
+        ChiselStage.emitVerilog({ new MyModule() }, true, true)
+      }
+    }
+    it("(?.d) A structurally identical but fully aligned monitor version of a bundle can easily be connected to") {
+      class Decoupled extends Bundle {
+        val valid = Bool()
+        val ready = Flipped(Bool())
+        val data = UInt(32.W)
+      }
+      class MyModule extends Module {
+        val in  = IO(Flipped(new Decoupled()))
+        val out = IO(new Decoupled())
+        val monitor = IO(Output(new Decoupled()))
```

Comment:
can you please show a version of this where the underlying type is not the same thing (`Decoupled` vs a `FlatDecoupled` where you don't use `Output` to strip the type, it just has the type manually. I would want to see that it is the same behavior